### PR TITLE
App Function

### PR DESCRIPTION
**Pull request by Krisnatagoras**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 101-app-function/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        "type": "string",
+        "defaultValue": "[resourceGroup().location]",
+        "metadata": {
+                "description": "Location for all resources."
+            }
+        }
+    },
+     "variables": {
+        "hostingPlanName": "[concat('hp', uniqueString(resourceGroup().name))]",
+        "storageAccountName": "[concat('sa',  uniqueString(resourceGroup().name))]"
```
Comment (bmoore-msft) [inline 101-app-function/azuredeploy.json]: this needs to be globally unique

File: 101-app-function/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        "location": {
+        "type": "string",
+        "defaultValue": "[resourceGroup().location]",
+        "metadata": {
+                "description": "Location for all resources."
+            }
+        }
+    },
+     "variables": {
+        "hostingPlanName": "[concat('hp', uniqueString(resourceGroup().name))]",
```
Comment (bmoore-msft) [inline 101-app-function/azuredeploy.json]: use a consistent naming pattern as discussed

File: 101-app-function/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "webAppName": {
+            "type": "string",
```
Comment (bmoore-msft) [inline 101-app-function/azuredeploy.json]: add a defaultValue as in other samples...

File: 101-app-function/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Web/sites",
+            "apiVersion": "2019-08-01",
+            "name": "[parameters('webAppName')]",
+            "kind": "functionapp,linux",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[concat('Microsoft.Web/serverfarms/', variables('hostingPlanName'))]",
```
Comment (bmoore-msft) [inline 101-app-function/azuredeploy.json]: remove concat and use resourceId

File: 101-app-function/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            "properties":{
+                "reserved": false
+            },
+            "sku": {
+                "Tier": "Dynamic",
+                "Name": "Y1"
+            }
+        },
+        {
+            "type": "Microsoft.Storage/storageAccounts",
```
Comment (bmoore-msft) [inline 101-app-function/azuredeploy.json]: what is this used for?

File: 101-app-function/metadata.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "itemDisplayName": "Function App.",
+  "description": "This template deploy a Function App. Also creates a Service Plan and a Storage Account.",
+  "summary": "This template deploy a Function App.",
```
Comment (bmoore-msft) [inline 101-app-function/metadata.json]: This template deploys an empty function app and a hosting plan.

File: 101-app-function/azuredeploy.json (hunk 1 above)
Comment (Krisnatagoras) [reply 101-app-function/azuredeploy.json]: Hey Brian, thanks for your comments.

So, in order to share the resources in the same group, changing to create the uniqueString from name to id I think that we will have a globally unique name, right?

File: 101-app-function/azuredeploy.json (hunk 2 above)
Comment (Krisnatagoras) [reply 101-app-function/azuredeploy.json]: "hostingPlanName": "[concat('hpn-', uniqueString(resourceGroup().id))]",

File: 101-app-function/azuredeploy.json (hunk 3 above)
Comment (Krisnatagoras) [reply 101-app-function/azuredeploy.json]: Done

File: 101-app-function/azuredeploy.json (hunk 4 above)
Comment (Krisnatagoras) [reply 101-app-function/azuredeploy.json]: Sure thing, I missed this one.
Done.

File: 101-app-function/metadata.json (hunk 6 above)
Comment (Krisnatagoras) [reply 101-app-function/metadata.json]: Thanks

File: 101-app-function/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,66 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "siteName": {
+            "type": "string",
+            "defaultValue": "[concat('FuncApp-', uniqueString(resourceGroup().name))]",
```
Comment (bmoore-msft) [inline 101-app-function/azuredeploy.json]: use resourceGroup().id as the uniqueString seed

File: 101-app-function/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        "location": {
+        "type": "string",
+        "defaultValue": "[resourceGroup().location]",
+        "metadata": {
+                "description": "Location for all resources."
+            }
+        }
+    },
+     "variables": {
+        "hostingPlanName": "[concat('hpn-', uniqueString(resourceGroup().id))]"
```
Comment (bmoore-msft) [inline 101-app-function/azuredeploy.json]: use resourceGroup().name as in your other samples, e.g.

concat('hpn-', resourceGroup().name)

File: 101-app-function/azuredeploy.json (hunk 7 above)
Comment (Krisnatagoras) [reply 101-app-function/azuredeploy.json]: Done

File: 101-app-function/azuredeploy.json (hunk 8 above)
Comment (Krisnatagoras) [reply 101-app-function/azuredeploy.json]: Also done!

**Pull request merged**